server-adapters/resteasy-undertow-cdi/src/main/java/dev/resteasy/embedded/server/UndertowCdiEmbeddedServer.java

```diff
@@ -0,0 +1,182 @@
+        final ContainerInstance container = deployment.getContainer();
+
+        // Ensure the RESTEasy deployment is started
+        EmbeddedServers.validateDeployment(deployment);
+
+        // Determine the servlet mapping name
+        String mapping = EmbeddedServers.checkContextPath(deployment);
+        if (!mapping.endsWith("/")) {
+            mapping += "/";
+        }
+        mapping = mapping + "*";
+
+        // Configure the RESTEasy Servlet
+        final ServletInfo resteasyServlet;
+        if (deploymentInfo.getServlets().containsKey("ResteasyServlet")) {
+            resteasyServlet = deploymentInfo.getServlets().get("ResteasyServlet");
+        } else {
+            resteasyServlet = Servlets.servlet("ResteasyServlet", HttpServlet30Dispatcher.class)
+                    .setAsyncSupported(true)
+                    .setLoadOnStartup(1)
+                    .addMapping(mapping);
+        }
+
+        if (!"/*".equals(mapping)) {
+            // Configure the mapping prefix for RESTEasy
+            final String prefix = mapping.substring(0, mapping.length() - 2);
+            resteasyServlet.addInitParam("resteasy.servlet.mapping.prefix", prefix);
+        }
+
+        // Check for context parameters
+        if (configuration.hasProperty(UndertowConfigurationOptions.CONTEXT_PARAMETERS)) {
+            final Object value = configuration.property(UndertowConfigurationOptions.CONTEXT_PARAMETERS);
+            if (value instanceof Map) {
+                for (Map.Entry<String, String> e : ((Map<String, String>) value).entrySet()) {
+                    deploymentInfo.addInitParameter(e.getKey(), e.getValue());
+                }
+            } else {
+                LogMessages.LOGGER.invalidProperty(UndertowConfigurationOptions.CONTEXT_PARAMETERS, value.getClass()
+                        .getName(), Map.class.getName());
+            }
+        }
+        // Determine the context path
+        final String contextPath = EmbeddedServers.checkContextPath(configuration.rootPath());
+        return deploymentInfo
+                // Set up deployment specific info
+                .setClassLoader(deployment.getApplication().getClass().getClassLoader())
+                .setContextPath(contextPath)
+                .setDeploymentName("RESTEasy-" + contextPath)
+                // Set up the RESTEasy Servlet
+                .addServletContextAttribute(ResteasyDeployment.class.getName(), deployment)
+                .addServlet(resteasyServlet)
+                // Configure Weld attributes and the listener
+                .addServletContextAttribute(Listener.CONTAINER_ATTRIBUTE_NAME, container)
+                .addServletContextAttribute(WeldServletLifecycle.BEAN_MANAGER_ATTRIBUTE_NAME, container.getBeanManager())
+                .addListener(Servlets.listener(Listener.class, new ImmediateInstanceFactory<>(Listener.using(container))));
```

Comment:
I think the configuration you do here is an overkill, i.e. you shouldn't need to add two attributes and a listener which, in this case, registers the same attribute again.
Basically, registering the `Listener` and using its `Listener.using()` method will automatically add the servlet context attribute, see [this Weld code](https://github.com/weld/core/blob/94595deb7dc8432a38e93a09ffb2556023ad3f3d/environments/servlet/core/src/main/java/org/jboss/weld/environment/servlet/Listener.java#L90-L93).
So in theory you could remove L178-179 and it should still work the same.